source/crud/crud.rst

```diff
@@ -1953,54 +1953,37 @@ in the collection prior to ``$merge`` being executed).
-attempt is needed.
-
-If it is not possible to augment the server selector and a pre-5.0 secondary is
-selected, drivers MAY invoke server selection a second time to obtain a primary.
-However, if the first attempt results in a server selection timeout, drivers
-MUST NOT make a second attempt and MUST propagate the original timeout error. If
-the topology type is Single, Sharded, or LoadBalanced, drivers MUST NOT make a
-second attempt (there is no benefit in doing so).
-
-Regardless of whether drivers employ a custom server selector or two-attempt
-approach, they MUST discern the *effective* read preference for the operation,
-which SHALL be used for specifying the
+or ``$merge`` and drivers never considered a read preference for the operation.
+As of ``featureCompatibilityVersion`` 4.4, secondaries can now execute pipelines
+with ``$out`` or ``$merge``. Since drivers do not track
+``featureCompatibilityVersion``, the decision to consider a read preference for
+such a pipeline will depend on the wire version(s) of the server(s) to which the
+driver is connected.
+
+If there are one or more available servers and one or more of those servers is
+pre-5.0 (i.e. wire version < 13), drivers MUST NOT use the available read
+preference and MUST instead select a server using a primary read preference.
+Otherwise, if there are either no available servers or all available servers are
+5.0+ (i.e. wire version >= 13), drivers MUST use the available read preference.
+
+Drivers SHOULD augment their
+`server selection algorithm <..../server-selection/server-selection.rst#server-selection-algorithm>`_
+such that this logic can be enforced within a single server selection attempt.
+
+If the driver is connected to a load balancer, it MUST consider the wire version
+of the backing server obtained from the connection handshake.
+
+Drivers MUST discern the read preference used to select a server for the
+operation, which SHALL be used for specifying the
 `$readPreference global command argument <../message/OP_MSG.rst#global-command-arguments>`_
 and
 `passing read preference to mongos and load balancers <../server-selection/server-selection.rst#passing-read-preference-to-mongos-and-load-balancers>`_
-(if applicable). The effective read preference SHALL be discerned as follows:
-
-- In the absence of an explicit or inherited read preference, the effective read
-  preference is ``{ "mode": "primary" }``.
-- If an explicit or inherited read preference results in selection of an
-  ineligible server and selection falls back to a primary, the effective read
-  preference is ``{ "mode": "primary" }``.
-- If a pre-5.0 mongos or a load balancer backing a pre-5.0 mongos is selected,
-  the effective read preference is ``{ "mode": "primary" }``.
-- If an explicit or inherited read preference is used and results in selection
-  of an eligible server (excluding a pre-5.0 mongos as discussed above), that is
-  the effective read preference.
+(if applicable).
```

Comment:
I don't think the original list is necessary. The first sentence in this paragraph is clear that the read preference used to select a server (either the available RP or the primary RP we fall back to) is what the driver must pass via `$readPreference`.